lib/screens/home/local_widget/province_card.dart

```diff
@@ -0,0 +1,238 @@
+  final TabController tabController;
+  final AutoScrollController scrollController;
+  final TbProvinceModel province;
+  final List<TbDistrictModel> district;
+
+  @override
+  Widget build(BuildContext context) {
+    return AutoScrollTag(
+      key: ValueKey(index),
+      controller: scrollController,
+      index: index,
+      child: Card(
+        elevation: 0,
+        margin: EdgeInsets.only(
+          top: ConfigConstant.margin2,
+          bottom: index == tabController.length - 1 ? 24 : 0,
+        ),
+        child: Column(
+          children: [
+            buildProvinceHeader(context),
+            Divider(height: ConfigConstant.margin1),
+            Container(
+              height: 56,
+              child: buildInfoCount(context),
+            ),
+            SizedBox(height: ConfigConstant.margin1),
+            Divider(height: 0),
+            buildTourPlaceListTile(
+              context: context,
+              title: 'តំបន់ទេសចរណ៍',
+              onTap: () {},
+            ),
+            Divider(height: 0),
+            buildDistictEpensionTile(context),
+          ],
+        ),
+      ),
+    );
+  }
+
+  ExpansionTile buildDistictEpensionTile(BuildContext context) {
+    bool isKhan = province.khan != 0;
+    return ExpansionTile(
+      initiallyExpanded: false,
+      title: Text(
+        isKhan ? 'ខណ្ឌ' : 'ស្រុក',
+        style: Theme.of(context)
+            .textTheme
+            .headline6
+            ?.copyWith(fontWeight: FontWeight.bold, color: Theme.of(context).primaryColor),
+      ),
+      subtitle: Text(
+        isKhan ? NumberHelper.toKhmer(province.khan) + ' ខណ្ឌ' : NumberHelper.toKhmer(province.srok) + ' ស្រុក',
+        style: Theme.of(context).textTheme.caption,
+      ),
+      trailing: Icon(Icons.keyboard_arrow_right),
+      children: List.generate(
+        district.length,
+        (index) {
+          String sangkat = NumberHelper.toKhmer(district[index].sangkat) + 'សង្កាត់';
+          String commune = NumberHelper.toKhmer(district[index].commune) + 'ឃុំ';
+          String village = NumberHelper.toKhmer(district[index].village) + 'ភូមិ';
+          bool isKrong = district[index].type == 'KRONG';
+          return Column(
+            children: [
+              Divider(height: 0),
+              (isKrong || isKhan)
+                  ? ListTile(
+                      title: Text((isKrong ? 'ក្រុង' : 'ខណ្ឌ') + (district[index].khmer ?? '')),
+                      subtitle: Text('$sangkat និង $village'),
+                      onTap: () {
+                        Navigator.restorablePushNamed(context, RouteConfig.DISTRICT);
+                      },
+                    )
+                  : ListTile(
+                      title: Text('ស្រុក' + (district[index].khmer ?? '')),
+                      subtitle: Text('$commune និង $villageភូមិ'),
+                      onTap: () {
+                        Navigator.restorablePushNamed(context, RouteConfig.DISTRICT);
```

Comment:
Good to use the new thing, never try it too. Can you tell me why we use`restorablePushNamed` instead of `Navigator.pushNamed()` here?

Reply:
That was an accident, I didn't notice that. I don't know what is `restorablePushNamed`.